DS/Assignment2/mycollectionclasses/MyArrayList.java

```diff
@@ -0,0 +1,132 @@
+package mycollectionclasses;
+
+//customised arraylist based on array
+public class MyArrayList {
+
+    private Object myArray[];
+    private int size;
+
+    public MyArrayList() {
+	myArray = new Object[10];
+	size = 0;
+    }
+
+    // Adding an item into array list and check if list is full, then resizing the arraylist
+    public void addElement(Object item) {
+	if (size < myArray.length) {
+	    myArray[size] = item;
+	    size++;
+	} else {
+	    Object newArray[] = new Object[myArray.length + 5];
+	    for (int i = 0; i < myArray.length; i++) {
+		newArray[i] = myArray[i];
+	    }
+	    newArray[size] = item;
+	    size++;
+	    myArray = newArray;
+	}
+    }
+
+    // Adding an item into array list at the given particular location i
+    public Object[] addElement(int index, Object item) {
+	for (int i = size; i > index; i--) {
+	    myArray[i] = myArray[i - 1];
+	}
+	myArray[index] = item;
```

Comment:
This insertion is only allowed when index is always less than size. See the implementation of List for more details.